corehq/apps/locations/resources/v0_6.py

```diff
@@ -35,4 +46,83 @@ def dehydrate(self, bundle):
             bundle.data['parent_location_id'] = ''
         bundle.data['location_type_name'] = bundle.obj.location_type.name
         bundle.data['location_type_code'] = bundle.obj.location_type.code
+        bundle.data['location_data'] = bundle.obj.metadata
+        return bundle
+
+    def obj_create(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if 'name' not in bundle.data or 'location_type_code' not in bundle.data:
+            raise BadRequest("'name' and 'location_type_code' are required fields.")
+        bundle.obj = SQLLocation(domain=domain)
+        self._update(bundle, domain, is_new_location=True)
         return bundle
+
+    def obj_update(self, bundle, **kwargs):
+        try:
+            bundle.obj = SQLLocation.objects.get(location_id=kwargs['location_id'], domain=kwargs['domain'])
+        except SQLLocation.DoesNotExist:
+            raise BadRequest(_("Could not find location with given ID on the domain."))
+        self._update(bundle, kwargs['domain'], is_new_location=False)
+        return bundle
+
+    def _update(self, bundle, domain, is_new_location=False):
+        data = bundle.data
+        # Invalid fields that might be common
+        if data.pop('location_type_name', False):
+            raise BadRequest(_('Location type name is not editable.'))
+        if data.pop('last_modified', False):
+            raise BadRequest(_('\"last_modified\" field is not editable.'))
```

Comment:
What does tastypie do by default if unrecognized properties are included?  Is there a way to make it error for _any_ unrecognized property?

Reply:
actually it looks like you do this already manually at the bottom of the `_update` function, so I think you could just drop these lines from here.